spec/abci/apps.md

```diff
@@ -111,31 +106,31 @@ an additional `Type` parameter is made available to the CheckTx function that
 indicates whether an incoming transaction is new (`CheckTxType_New`), or a
 recheck (`CheckTxType_Recheck`).
 
-Finally, the mempool will unlock and new transactions can be processed through CheckTx again.
+Finally, after re-checking transactions in the mempool, Tendermint will unlock 
+the mempool connection. New transactions are once again able to be processed through CheckTx.
+
+Note that CheckTx is just a weak filter to keep invalid transactions out of the block chain. 
+CheckTx doesn't have to check everything that affects transaction validity; the
+expensive things can be skipped.  It's weak because a Byzantine node doesn't 
+care about CheckTx; it can propose a block full of invalid transactions if it wants.
 
-Note that CheckTx doesn't have to check everything that affects transaction validity; the
-expensive things can be skipped. In fact, CheckTx doesn't have to check
-anything; it might say that any transaction is a valid transaction.
-Unlike DeliverTx, CheckTx is just there as
-a sort of weak filter to keep invalid transactions out of the blockchain. It's
-weak, because a Byzantine node doesn't care about CheckTx; it can propose a
-block full of invalid transactions if it wants.
+-- what other line of defense do we have against invalid TX? TM itself i'm assuming? --
 
 #### Replay Protection
 
 To prevent old transactions from being replayed, CheckTx must implement
 replay protection.
 
+-- is this implementation specific or should it be included in the spec? -- 
 Tendermint provides the first defense layer by keeping a lightweight
-in-memory cache of 100k (`[mempool] cache_size`) last transactions in
-the mempool. If Tendermint is just started or the clients sent more than
-100k transactions, old transactions may be sent to the application. So
+in-memory cache of recent transactions in the mempool. If Tendermint is just 
```

Comment:
In strikes me that information about in-process caching should really be moved over to the documentation in the Go repo. Maintaining a cache is somewhat independent of the specification of the program and protocol.